bc_obps/registration/api/operation.py

```diff
@@ -303,27 +317,31 @@ def update_operation_status(request, operation_id: int, payload: OperationUpdate
     user: User = request.current_user
     status = Operation.Statuses(payload.status)
     operation.status = status
-    if status in [Operation.Statuses.APPROVED, Operation.Statuses.DECLINED]:
-        operation.verified_at = datetime.now(pytz.utc)
-        operation.verified_by = user
-        if status == Operation.Statuses.APPROVED:
-            try:
-                operation.generate_unique_boro_id()
-                # approve the operator if it's not already approved (the case for imported operators)
-                operator: Operator = operation.operator
-                if operator.status != Operator.Statuses.APPROVED:
-                    operator.status = Operator.Statuses.APPROVED
-                    operator.is_new = False
-                    operator.verified_at = datetime.now(pytz.utc)
-                    operator.verified_by = user
-                    operator.save(update_fields=["status", "is_new", "verified_at", "verified_by"])
-                    operator.set_create_or_update(modifier=user)
-            except Exception as e:
-                return 400, {"message": str(e)}
     try:
-        operation.save()
-        operation.set_create_or_update(modifier=user)
+        with transaction.atomic():
+            if status in [Operation.Statuses.APPROVED, Operation.Statuses.DECLINED]:
+                operation.verified_at = datetime.now(pytz.utc)
+                operation.verified_by = user
+                if status == Operation.Statuses.APPROVED:
+                    try:
+                        operation.generate_unique_boro_id()
+                        # approve the operator if it's not already approved (the case for imported operators)
+                        operator: Operator = operation.operator
+                        if operator.status != Operator.Statuses.APPROVED:
+                            operator.status = Operator.Statuses.APPROVED
+                            operator.is_new = False
+                            operator.verified_at = datetime.now(pytz.utc)
+                            operator.verified_by = user
+                            operator.save(update_fields=["status", "is_new", "verified_at", "verified_by"])
+                            operator.set_create_or_update(modifier=user)
+                    except Exception as e:
+                        return 400, {"message": str(e)}
+
+                operation.save()
+                operation.set_create_or_update(modifier=user)
+
+                return 200, operation
```

Comment:
The backend test was additionally failing because the new try/except blocks returned 400 Exception instead of a ValidationError. I changed the test because I think that's what we want, but if not, let me know, thanks!